packages/neon_framework/lib/src/widgets/dialog.dart

```diff
@@ -523,3 +555,395 @@ class NeonUnifiedPushDialog extends StatelessWidget {
+                      user_status.ClearAt(
+                        (b) => b
+                          ..type = user_status.ClearAt_Type.endOf
+                          ..time = ($int: null, clearAtTimeType: user_status.ClearAtTimeType.week),
+                      ),
+                    ]
+                        .map(
+                          (clearAt) => DropdownMenuItem(
+                            value: clearAt,
+                            enabled: selected == null || clearAt != selected,
+                            child: Text(clearAt.formatRelative(context)),
+                          ),
+                        )
+                        .toList(),
+                  );
+                },
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+
+    return NeonDialog(
+      content: body,
+      automaticallyShowCancel: false,
+      actions: [
+        NeonDialogAction(
+          isDestructiveAction: true,
+          onPressed: () {
+            bloc.clearMessage();
+          },
+          child: Text(NeonLocalizations.of(context).userStatusActionClear),
+        ),
+        NeonDialogAction(
+          isDefaultAction: true,
+          onPressed: () {
+            Navigator.of(context).pop();
+          },
+          child: Text(NeonLocalizations.of(context).actionClose),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildOnlineStatuses() {
+    final onlineStatuses = {
+      'online': (
+        ['online'],
+        NeonLocalizations.of(context).userStatusOnline,
+        null,
+      ),
+      'dnd': (
+        ['dnd'],
+        NeonLocalizations.of(context).userStatusDoNotDisturb,
+        NeonLocalizations.of(context).userStatusDoNotDisturbDescription
+      ),
+      'away': (
+        ['away'],
+        NeonLocalizations.of(context).userStatusAway,
+        null,
+      ),
+      'invisible': (
+        ['invisible', 'offline'],
+        NeonLocalizations.of(context).userStatusInvisible,
+        NeonLocalizations.of(context).userStatusInvisibleDescription
+      ),
+    };
+
+    return ResultBuilder.behaviorSubject(
+      subject: bloc.status,
+      builder: (context, result) {
+        final onlineStatusTiles = onlineStatuses.entries.map(
+          (onlineStatus) {
+            final (types, title, subtitle) = onlineStatus.value;
+            return Card(
+              shape: types.contains(result.data?.status) ? selectedBorder : null,
+              child: InkWell(
+                borderRadius: const BorderRadius.all(Radius.circular(12)),
+                onTap: () {
+                  bloc.setStatusType(onlineStatus.key);
+                },
+                child: Center(
+                  child: ListTile(
+                    leading: NeonServerIcon(icon: 'user-status-${onlineStatus.key}'),
+                    title: Text(title),
+                    subtitle: subtitle != null ? Text(subtitle) : null,
+                  ),
+                ),
+              ),
+            );
```

Comment:
```suggestion
            return Card(
              clipBehavior: Clip.hardEdge,
              shape: types.contains(result.data?.status) ? selectedBorder : null,
              child: ListTile(
                leading: NeonServerIcon(icon: 'user-status-${onlineStatus.key}'),
                title: Text(title),
                subtitle: subtitle != null ? Text(subtitle) : null,
                onTap: () {
                  bloc.setStatusType(onlineStatus.key);
                },
              ),
            );
```

Just set the clip behavior on the card if you want to avoid the splash overflowing.